client/lib/domains/dns/index.js

```diff
@@ -126,7 +130,57 @@ function getFieldWithDot( field ) {
 	return ( typeof field === 'string' && field.match( /^([a-z0-9-_]+\.)+\.?[a-z]+$/i ) ) ? field + '.' : field;
 }
 
-module.exports = {
-	validateAllFields,
-	getNormalizedData
+function isWpcomRecord( record ) {
+	return startsWith( 'wpcom:', record.id );
+}
+
+function filterRootARecords( domain, records ) {
+	return filter( records, {
+		type: 'A',
+		name: `${domain}.`
+	} );
+}
+
+function removeDuplicateWpcomRecords( domain, records ) {
+	const wpcomRecord = find( filterRootARecords( domain, records ), isWpcomRecord ),
+		customRecord = find( records, { type: 'A', name: '' } );
+
+	let result;
```

Comment:
I think so - `eslint` doesn't show any warning for such a code here. And I don't see any rules `.eslintrc.js` that would trigger this warning. AFAIK, `eslint` also prints (at the end of the warning) the name of the rule that triggered it - you can check what is it and if it's in Calypso's `.eslintrc.js` or is that something else conflicting with it?